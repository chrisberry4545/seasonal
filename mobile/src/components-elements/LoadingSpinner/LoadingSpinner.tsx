import React, { FC } from 'react';
import { ActivityIndicator, ViewStyle } from 'react-native';

export const LoadingSpinner: FC<{
  size?: 'small' | 'large',
  style?: ViewStyle
}> = ({
  size = 'large',
  style
} = {}) => (
  <ActivityIndicator size={size} style={style} />
);
