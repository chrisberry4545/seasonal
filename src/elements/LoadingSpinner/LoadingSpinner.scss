@import './../../styles/settings';
@import './../../styles/tools';

$loading-spinner__dot--time-between-starts: 0.23s;
$loading-spinner__dot--starting-scale: 0.5;

@keyframes growAnimation {
    0%, 80%, 100% {
        transform: scale($loading-spinner__dot--starting-scale);
        background: darken($loading-spinner-color, 30%);
    }
    20% {
        transform: scale(1);
        background: $loading-spinner-color;
    }
}

.c-loading-spinner {
  display: inline-block;
}

  .c-loading-spinner__dot {
    display: inline-block;
    width: 12px;
    height: 12px;
    background-color: $loading-spinner-color;
    border-radius: 50%;
    animation-name: growAnimation;
    animation-duration: 1.75s;
    animation-iteration-count: infinite;

    &:nth-child(1) {
      animation-delay: $loading-spinner__dot--time-between-starts * 0;
    }

    &:nth-child(2) {
      animation-delay: $loading-spinner__dot--time-between-starts * 1;
    }

    &:nth-child(3) {
      animation-delay: $loading-spinner__dot--time-between-starts * 2;
    }

    &:nth-child(4) {
      animation-delay: $loading-spinner__dot--time-between-starts * 3;
    }
  }
