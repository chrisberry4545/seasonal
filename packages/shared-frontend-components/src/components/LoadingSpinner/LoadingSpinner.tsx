import React, { FC } from 'react';

import './LoadingSpinner.scss';

export const LoadingSpinner: FC<{}> = () => (
  <div className='c-loading-spinner' data-e2e='loading-spinner'>
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
  </div>
);
