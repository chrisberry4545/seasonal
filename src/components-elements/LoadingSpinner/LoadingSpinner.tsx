import React from 'react';

import './LoadingSpinner.scss';

export const LoadingSpinner = () => (
  <div className='c-loading-spinner'>
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
    <div className='c-loading-spinner__dot' />
  </div>
);
