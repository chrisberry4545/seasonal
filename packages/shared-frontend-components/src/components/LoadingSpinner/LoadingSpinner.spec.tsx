import React from 'react';
import { LoadingSpinner } from './LoadingSpinner';
import { shallow, ShallowWrapper } from 'enzyme';

describe('<LoadingSpinner />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<LoadingSpinner />);
  });

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());
});
