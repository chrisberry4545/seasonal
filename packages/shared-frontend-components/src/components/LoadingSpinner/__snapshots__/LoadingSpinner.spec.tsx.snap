// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> renders correctly 1`] = `
<div
  className="c-loading-spinner"
  data-e2e="loading-spinner"
>
  <div
    className="c-loading-spinner__dot"
  />
  <div
    className="c-loading-spinner__dot"
  />
  <div
    className="c-loading-spinner__dot"
  />
  <div
    className="c-loading-spinner__dot"
  />
</div>
`;
